draft-ietf-httpbis-connect-tcp.md

```diff
@@ -44,9 +44,9 @@ HTTP/3 uses a UDP transport, so it cannot be forwarded using the pre-existing CO
 
 ## Problems
 
-Classic HTTP CONNECT proxies are identified by an origin.  The proxy does not have a path of its own.  This prevents any origin from hosting multiple distinct proxy services.
+Classic HTTP CONNECT proxies are identified by a host, port, and scheme.  The proxy does not have a path of its own.  This prevents any origin from hosting multiple distinct proxy services.
 
-Ordinarily, HTTP allows multiple origin hostnames to share a single server IP address and port number (i.e., virtual-hosting), by specifying the applicable hostname in the "Host" or ":authority" header field.  However, classic HTTP CONNECT proxies use these fields to indicate the CONNECT request's destination ({{?RFC9112, Section 3.2.3}}), leaving no way to determine the proxy's origin from the request.  As a result, classic HTTP CONNECT proxies cannot be deployed using virtual-hosting, nor can they apply the usual defenses against server port misdirection attacks (see {{Section 7.4 of ?RFC9110}}).
+Ordinarily, HTTP allows multiple origin hostnames to share a single server IP address and port number (i.e., virtual-hosting), by specifying the applicable hostname in the "Host" or ":authority" header field.  However, classic HTTP CONNECT proxies use these fields to indicate the CONNECT request's destination ({{?RFC9112, Section 3.2.3}}), leaving no way to distinguish multiple proxies on one server.  As a result, classic HTTP CONNECT proxies cannot be deployed using virtual-hosting, nor can they apply the usual defenses against server port misdirection attacks (see {{Section 7.4 of ?RFC9110}}).  The absence of an explicit origin for the proxy also creates ambiguity about the use of origin-scoped response header fields such as Alt-Svc {{?RFC7838}} and Strict-Transport-Security {{?RFC6797}}.
```

Comment:
```suggestion
Ordinarily, HTTP allows multiple origin hostnames to share a single server IP address and port number (i.e., virtual-hosting), by specifying the applicable hostname in the "Host" or ":authority" header field.  However, classic HTTP CONNECT proxies use these fields to indicate the CONNECT request's destination ({{?RFC9112, Section 3.2.3}}), leaving no way to distinguish multiple proxy origins on one server.  As a result, classic HTTP CONNECT proxies cannot be deployed using virtual-hosting, nor can they apply the usual defenses against server port misdirection attacks (see {{Section 7.4 of ?RFC9110}}).  The absence of an explicit origin for the proxy also creates ambiguity about the use of origin-scoped response header fields such as Alt-Svc {{?RFC7838}} and Strict-Transport-Security {{?RFC6797}}.
```

Reply:
I actually prefer @bemasc's text here. The term "Proxy origin" sounds confusing to me, as origin is the CONNECT destination in case of traditional CONNECT. "Proxy" is more succinct and serves the purpose.

Reply:
Yeah "proxy" is a term defined in https://www.rfc-editor.org/rfc/rfc9110.html#section-3.7-5, I think we might want to refer to that definition, considering that this is an HTTP extension.

Regarding how we talk, I think things might become clearer if we talk about "HTTP proxies" here. HTTP proxies are identified by host, port, and use of TLS. Traditional CONNECT is just one way of using the HTTP proxies.

Would could point that out, then explain that we are creating a way to establish a TCP tunnel "through HTTP origin."

Reply:
OK, I've added a reference to that section and clarified that we mean "HTTP proxy".